uint128_t.cpp

```diff
@@ -20,6 +20,46 @@ uint128_t::uint128_t(uint128_t && rhs)
     }
 }
 
+uint128_t::uint128_t(const char *s)
```

Comment:
Why a `char *`? Why not `std::string`?

Reply:
If `char *` is preferable to `std::string`, maybe have the `std::string` function call the `char *` version?